sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStore.java

```diff
@@ -433,6 +440,92 @@ public void clearProperty(final String key) throws StateStoreException {
         }
     }
 
+    // Read/Write task metadata
+
+    /**
+     * Stores the goal state override status of a particular Task. The {@link TaskInfo} for this exact task MUST have
+     * already been written via {@link #storeTasks(Collection)} beforehand.
+     *
+     * @throws StateStoreException in the event of a storage error
+     */
+    public void storeGoalOverrideStatus(String taskName, GoalStateOverride.Status status)
+            throws StateStoreException {
+        try {
+            Map<String, byte[]> values = new TreeMap<>();
+
+            if (GoalStateOverride.Status.INACTIVE.equals(status)) {
+                // Mark inactive state by clearing any override bits.
+                values.put(getGoalOverridePath(taskName), null);
+                values.put(getGoalOverrideStatusPath(taskName), null);
+            } else {
+                values.put(getGoalOverridePath(taskName),
+                        status.target.getSerializedName().getBytes(StandardCharsets.UTF_8));
+                values.put(getGoalOverrideStatusPath(taskName),
+                        status.progress.getSerializedName().getBytes(StandardCharsets.UTF_8));
+            }
+            persister.setMany(values);
+        } catch (PersisterException e) {
+            throw new StateStoreException(e);
+        }
+    }
+
+    /**
+     * Retrieves the goal state override status of a particular task. A lack of override will result in a
+     * {@link GoalOverrideStatus} with {@code override=NONE} and {@code state=NONE}.
+     *
+     * @throws StateStoreException in the event of a storage error
+     */
+    public GoalStateOverride.Status fetchGoalOverrideStatus(String taskName) throws StateStoreException {
+        try {
+            String goalOverridePath = getGoalOverridePath(taskName);
+            String goalOverrideStatusPath = getGoalOverrideStatusPath(taskName);
+            Map<String, byte[]> values = persister.getMany(Arrays.asList(goalOverridePath, goalOverrideStatusPath));
+            byte[] nameBytes = values.get(goalOverridePath);
+            byte[] statusBytes = values.get(goalOverrideStatusPath);
+            if (nameBytes == null && statusBytes == null) {
+                // Cleared override bits => Inactive state
+                return GoalStateOverride.Status.INACTIVE;
+            } else if (nameBytes == null || statusBytes == null) {
+                throw new StateStoreException(Reason.LOGIC_ERROR,
```

Comment:
Do we definitely want to throw here?  Maybe we should return `INACTIVE` and so fallback to original behavior.  Then a user could retry the override, I think.  It would be nice for the end-user to be able to clear/retry overrides.

Reply:
Sounds reasonable, in theory shouldn't happen but doesn't hurt to have a reasonable fallback here.